### ISSUE TITLE
|Bug| y_malloc ?

### ISSUE THREAD

**Issue by HxTheDash** [pawn-lang/YSI-Includes#91]
Hi guys,

For a long time I've been having trouble with Y_Inline.

The Bug happens when I restart my server after the restart no function to work with the y_inline works.

I made some debug in Y_Inline until I came to the conclusion that the function Y_Malloc is causing conflict.

``` Pawn

stock bool: Callback_Get (callback: name, ret [E_CALLBACK_DATA] expect [] = "", bool: remote = false);
```

Within this function I did some debugs to "try" to find where this bug was coming.

I noticed that this line **Alloc:a = malloc (stack);**  was getting the value of ZERO (0) after restart of my server.

``` Pawn
Alloc: a = malloc (stack); // Allocate space closure.
printf ( "alloc:% d", _:a);
if (a)
{

```

After some testing, resolve to take the Y_malloc of YSI 3.1 and adapt YSI 4.0, I concluded that the bug really was in Y_Malloc.

I can is wrong or not, but this bug has been bothering many people.

**Comment by Y-Less** [comment]
The problem is, that despite mulitple people reporting this issue, and all being asked the same questions, no-one has yet actually provided a reproduction case - i.e. a small mode that consistently demonstrates the bug.  YSI has lots of automated tests, including ones for y_inline and y_malloc.  Every time I run those, even with server restarts involved, they pass.  So clearly there is something more that your modes do that the tests don't, and that is what causes the problem.  With a minimal reproduction case it would be quite easy to narrow down what that thing is and fix it.  Without one, I can't at all, regardless of the number of times that it gets reported.

**Comment by Y-Less** [comment]
Although this is the closest anyone has got to something more useful than "it doesn't work, please fix".

**Comment by Y-Less** [comment]
Hopefully fixed by #101 when using `YSI_NO_HEAP_MALLOC` as a compile-time setting.  This reverts y_malloc to the old style of using a huge array instead of the heap, but will increase compile times and file sizes.
